AzureCalling/Controllers/IntroViewController.swift

```diff
@@ -0,0 +1,265 @@
+        stackView.pinToBottom()
+    }
+
+    private func layoutMainContainer() {
+        let titleLabel = FluentUI.Label(style: .title1)
+        titleLabel.text = "Video calling sample"
+
+        let builtWithLabel = FluentUI.Label(style: .body)
+        builtWithLabel.text = "Built with"
+        builtWithLabel.directionalLayoutMargins.top = 24
+
+        let acsImageLabel = FluentUI.Label(style: .button1)
+        acsImageLabel.text = "Azure Communication Services"
+        acsImageLabel.textColor = ThemeColor.primary
+
+        let acsLabelStack = UIStackView(arrangedSubviews: [
+            UIImageView(image: UIImage(named: "acsLogo")),
+            acsImageLabel
+        ])
+        acsLabelStack.spacing = 8
+
+        let labelContainer = UIView()
+        labelContainer.translatesAutoresizingMaskIntoConstraints = false
+        labelContainer.addSubview(builtWithLabel)
+        labelContainer.addSubview(acsLabelStack)
+        builtWithLabel.centerHorizontallyInContainer()
+        acsLabelStack.centerHorizontallyInContainer()
+
+        labelContainer.layer.cornerRadius = 12
+        labelContainer.backgroundColor = ThemeColor.lightSurfacesPrimary
+        NSLayoutConstraint.activate(
+            NSLayoutConstraint.constraints(
+                withVisualFormat: "V:|-24-[builtWith]-[labelStack]-24-|",
+                metrics: nil,
+                views: ["builtWith": builtWithLabel,
+                        "labelStack": acsLabelStack]
+            )
+        )
+
+        let stackView = UIStackView(arrangedSubviews: [titleLabel, labelContainer])
+        labelContainer.expandHorizontallyInSuperView()
+
+        stackView.axis = .vertical
+        stackView.alignment = .center
+        stackView.spacing = 32
+        view.addSubview(stackView)
+
+        stackView.expandHorizontallyInSuperView(withEqualMargin: 16)
+        stackView.centerVerticallyInContainer()
+    }
+
+    private func layoutTopBar() {
+        let container = UIView()
+        topBar = container
+        view.addSubview(container)
+        topBar.pinToTop()
+        topBar.expandHorizontallyInSuperView()
+        topBar.addSubview(signOutButton)
+        signOutButton.expandVerticallyInSuperView()
+        signOutButton.pinToRight()
+
+        userAvatar = MSFAvatar(style: .default, size: .small)
+        userDisplayName = FluentUI.Label(style: .title2, colorStyle: .primary)
+
+        let userDetails = UIStackView(arrangedSubviews: [
+            userAvatar.view,
+            userDisplayName
+        ])
+        userDetails.spacing = 8
+
+        topBar.addSubview(userDetails)
+        userDetails.pinToLeft(withMargin: 16)
+    }
+
+    // MARK: - Authentication State Handling
+    private func handleAuthState() {
+        userAvatar.state.image = authHandler.userAvatar
+        userAvatar.state.primaryText = authHandler.userDisplayName
+        userDisplayName.text = authHandler.userDisplayName
+
+        switch authHandler.authStatus {
```

Comment:
For the scenario that the user is previously logged in, the authStatus is .unauthorized so the "Sign in" button will appear and after tap on the Sign in button, authHandler.login will find a result and automatically change the view to "Start call" and "Join call". We should somehow check authHandler.login earlier to avoid the extra step of tapping the Sign in button.

Reply:
Yeah, we could check for an account at the load of the view controller and if so, silently log in.